SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigImpl.c

```diff
@@ -4651,8 +4690,49 @@ SecureBootCallback (
       // When entering SecureBoot OPTION Form
       // always close opened file & free resource
       //
-      if ((QuestionId == KEY_SECURE_BOOT_PK_OPTION) ||
-          (QuestionId == KEY_SECURE_BOOT_KEK_OPTION) ||
+      if ((QuestionId == KEY_SECURE_BOOT_PK_OPTION)){
+        CloseEnrolledFile (Private->FileContext);
+
+        GetVariable2 (EFI_PLATFORM_KEY_NAME, &gEfiGlobalVariableGuid, (VOID **)&PkList, NULL);
+        if (PkList != NULL) {
+          // Extract the PK certificate from the list
+          PkCert = (EFI_SIGNATURE_DATA *)((UINT8 *)PkList + sizeof (EFI_SIGNATURE_LIST) + PkList->SignatureHeaderSize);
+          UINTN cnSize = 0;
+          X509GetCommonName(
+            (UINT8*)PkCert->SignatureData,
+            (UINTN)PkList->SignatureSize,
+            NULL,
+            &cnSize);
+
+          PkCN8 = AllocateZeroPool(cnSize*sizeof(CHAR8));
+          if (PkCN8 == NULL) {
+            return EFI_OUT_OF_RESOURCES;
+          }
+          PkCN16 = AllocateZeroPool(cnSize*sizeof(CHAR16));
+          if (PkCN16 == NULL) {
+            return EFI_OUT_OF_RESOURCES;

```

Comment:
Returning here will result in PkCN8 and PkList not being freed

Reply:
Can such problems be detected by tools such as e.g. coverity?

Does upstream edk2 use any similar tools in Ci we could include here?

Reply:
tianocore edk2 uses CodeQL for checks and I'm not sure if these cases are detected. Coverity might be easier to implement

Reply:
[Fixed](https://github.com/Dasharo/edk2/pull/149/files#diff-49fdb815b38763cce7b30e2aea02c17111bc51206950ff1d21e0073ebd0cb1d3R4728)